### DISCUSSION TITLE
Having issues setting up using Vagrant

### DISCUSSION THREAD

**Discussion by sh-how** [docker-mailserver/docker-mailserver#2787] [Q&A]
# Having issues setting up using Vagrant

I have no issues with the mail server when I use docker-compose method of setting it up but I would like to setup the mail server using Vagrant. The vagrant box shows that the container is running and I am able to ping it. However, I cannot access any of its services e.g., SMTP via telnet as it shows ```telnet: Unable to connect to remote host: Connection refused```

## Commands ran

```
git clone --recurse-submodules https://github.com/docker-mailserver/docker-mailserver
cd docker-mailserver
./setup  email add test@example.com password
vagrant init
vagrant up
```

## Vagrant file
```
Vagrant.configure("2") do |config|
  config.vm.define "docker" do |docker|
    docker.vm.provider "docker" do |d|
      d.build_dir = "."
      d.cmd = ["tail", "-f", "/dev/null"]
    end
  end
end
```

I have tried allow port 25 using ```ufw allow 25``` but the connection is still getting refused. Running ```docker ps``` shows that the container is up

```
2699e3d7a460   3c7e7d7e086a   "/usr/bin/dumb-init …"   6 seconds ago   Up 4 seconds   25/tcp, 110/tcp, 143/tcp, 465/tcp, 587/tcp, 993/tcp, 995/tcp, 4190/tcp   docker-mailserver_docker_1663737404
```

```
docker exec 2699e3d7a460 ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
86: eth0@if87: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default 
    link/ether 02:42:ac:11:00:02 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 172.17.0.2/16 brd 172.17.255.255 scope global eth0
       valid_lft forever preferred_lft forever

telnet 172.17.0.2 25
Trying 172.17.0.2...
telnet: Unable to connect to remote host: Connection refused
```

Another question I have is whether the email setup process is the same when setting up the mail server with Vagrant

**Comment by georglauterbach** [comment]
I migrated this to a discussion, I hope you don't mind. The issue tracker is mainly for supported features and maintainers' time is sparse :) Hopefully, someone in the discussions site can help you.
